src/form/input/Input.tsx

```diff
@@ -49,32 +48,77 @@ export type InputProps = Omit<
   customClassName?: string;
   inputContainerRef?: React.RefObject<HTMLDivElement>;
   onChange: React.ReactEventHandler<HTMLInputElement>;
-  maxFractionDigits?: number;
+  localizationOptions?: {
+    shouldFormatToLocaleString?: boolean;
+    locale?: string;
+    maximumFractionDigits?: number;
+  };
 };
 
+/* eslint-disable complexity */
 function Input(props: InputProps) {
   const {
     testid,
+    value,
     type = "text",
     isDisabled,
     hasError,
     customClassName,
     leftIcon,
     rightIcon,
+    localizationOptions = {},
     role,
     autoComplete = "off",
     autoCorrect = "off",
     inputContainerRef,
     onChange,
-    maxFractionDigits = 0,
     ...rest
   } = props;
+  const {
+    shouldFormatToLocaleString = false,
+    locale,
+    maximumFractionDigits = 0
+  } = localizationOptions;
+  const {DECIMAL_NUMBER_SEPARATOR} = getNumberSeparators(locale);
   const inputContainerClassName = classNames("input-container", customClassName);
   const inputClassName = classNames("input", {
     "input--is-disabled": isDisabled,
     "input--has-error": hasError
   });
   const isNumberInput = type === "number";
+  let finalValue = value;
+
+  if (isNaN(Number(value))) {
+    finalValue = "";
+  }
```

Comment:
In the current version, `parseNumber` returns `NaN` that are not numbers. I've tried `event.preventDefault()` in `handleChange` but it's not working. (cc: @edizcelik) You can take a look [here](https://stackoverflow.com/questions/32298064/preventdefault-not-working-in-on-input-function). 
Therefore, I thought we can handle like that but I'm not sure it's good solution.

Reply:
We need to handle this within `handleChange`. If `formattedNewValue` is NaN after `parseNumber` is called, then we can preventDefault. we should be able to use preventDefault for onChange handlers, not sure why it isn't working. If it is really not working, I assume we can revert back to the latest `value` (or the computed finalValue) that is provided as a prop .

Reply:
I've investigated but it seems there is no way use `event.preventDefault()` for input event. I've tried for `SyntheticEvent`, `KeyboardEvent`, `ChangeEvent` but `HTMLInputElement` have not `key` property for use `preventDefault`. 
I guess as you said, we should use previous state.

Reply:
We can just not call the `onChange` handler in this case so that the value wouldn't be updated. Would that work ?

Reply:
I've tried a similar one but didn't work. I guess really correct a RegExp will solve this. Even it's working now but in dynamic format causes some weird behaviors.

Reply:
Does the check on line:91 work ? If so, you can just do the same within `handleChange`

Reply:
It's working only in one case that hasn't `maximumFractionDigits` prop.
I believe that if we can work `^-?([0-9]*,*)*\\.?[0-9]*$` Regex as dynamically, it will work. But somehow it's not working. I guess I make a mistake in some operators.

Shouldn't be like this?
```js
const NUMERALS = "[0123456789]"; // dynamically
const NUMBER_WITH_THOUTHOUSANDTH_AND_DECIMAL_POINT_REGEX = new RegExp(
 `^-?(${NUMERALS}|[0-9]*${THOUSANDTHS_SEPARATOR}*)*\\${DECIMAL_SEPARATOR}?${NUMERALS}|[0-9]*$`
);
```